neuro_san/internals/interfaces/agent_registry.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+from typing import Any
+from typing import Dict
+
+from neuro_san.internals.interfaces.front_man import FrontMan
+
+
+class AgentRegistry:
+    """
+    Interface that describes how the session-level of the system interacts with
+    a registry of agents.
+    """
```

Comment:
This might change a bit in my next level of refactor after this.
That will include having:
*  the tool_factories stuff actually yield AgentNetworks (instances that can only look at config)  and
* an AgentRegistry is created with a new copy of an AgentNetwork for each request.